### PR TITLE
allows mix precision compilation

### PR DESCRIPTION
**Pull request by JosephMouallem**
Updates real kind of some variables given that their corresponding routines are called from the full coupler which is compiled in 64bit.

### CODE REVIEW

**Comment by JosephMouallem** [comment]
To work with:
https://github.com/NOAA-GFDL/SHiELD_build/pull/46
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/360

File: atmos_shared/atmos_tracer_driver.F90 [hunk 1]
```diff
@@ -19,6 +19,7 @@
 !***********************************************************************
 
 module atmos_tracer_driver_mod
+#include <fms_platform.h>
```
Comment (bensonr) [inline atmos_shared/atmos_tracer_driver.F90:22]: I don't think we need both fms_platform.h and the use_module platform_mod.

File: atmos_shared/atmos_global_diag.F90 [hunk 2]
```diff
@@ -1,10 +1,12 @@
 module atmos_global_diag_mod
+#include <fms_platform.h>
```
Comment (bensonr) [inline atmos_shared/atmos_global_diag.F90:2]: I don't think we need both fms_platform.h and the use_module platform_mod.

File: atmos_shared/atmos_tracer_driver.F90 (hunk 1 above)
Comment (JosephMouallem) [reply atmos_shared/atmos_tracer_driver.F90]: Good catch, removed.

**Review by laurenchilutti — APPROVED** [review]

**Comment by laurenchilutti** [comment]
Once Rusty completes hiw review of this, all 3 PRs will be fully reviewed and I can merge

**Review by bensonr — APPROVED** [review]

**Pull request merged**